src/coredump/coredump.c

```diff
@@ -679,26 +678,26 @@ static int get_mount_namespace_leader(pid_t pid, pid_t *container_pid) {
 
         for (;;) {
                 ino_t parent_mntns;
+                pid_t ppid;
 
-                r = get_process_ppid(cpid, &ppid);
+                r = get_process_ppid(pid, &ppid);
+                if (r == -EADDRNOTAVAIL) /* Reached the top (i.e. typically PID 1, but could also be a process
+                                          * whose parent is not in our pidns) */
+                        return -ENOENT;
                 if (r < 0)
                         return r;
 
                 r = get_process_ns(ppid, "mnt", &parent_mntns);
                 if (r < 0)
                         return r;
 
-                if (proc_mntns != parent_mntns)
-                        break;
-
-                if (ppid == 1)
-                        return -ENOENT;
+                if (proc_mntns != parent_mntns) {
+                        *ret = ppid;
+                        return 0;
+                }
 
-                cpid = ppid;
+                pid = ppid;
         }
-
-        *container_pid = ppid;
-        return 0;
```

Comment:
Doesn't the compiler complain about the lack of `return`? Or is it smart enough to detect the infinite loop with no breaks?

Reply:
yes, it is smart enough for that. CI will prove that... ;-)